### ISSUE TITLE
Allow custom providers

### ISSUE THREAD

**Issue by cesarandreu** [simov/grant#15]
Any plans on allowing custom providers? 

In config.init you could merge the passed in options into the result.

**Comment by simov** [comment]
I think that's doable, although you'll have to add every provider option by yourself in this case. I'm giving you example:

```js
"myprovider": {
    "authorize_url": "https://www.myprovider.com/authorize",
    "access_url": "https://api.myprovider.com/auth/token",
    "oauth": 2,
    "scope_delimiter": " ",
   // any other options that you normally add follows
  }
```

And these are currently not documented, but I can add some docs for them too. Without these options your custom provider won't work.

**Comment by cesarandreu** [comment]
I know you'd have to add those extra fields.

In the meantime I just used one of the existing providers and overwrote the values:

```js
{
  facebook: {
    name: 'foo'
    authorize_url: '...',
    access_url: '...',
    oauth: 2,
    ...
  }
}
```

A bit hacky, but it works :D.

**Comment by simov** [comment]
That actually reminds me of something else that I can add, also I'm going to implement this feature in the next few days :+1:

**Comment by simov** [comment]
Just published v3.2.0 there you can have [custom provider](https://github.com/simov/grant#custom-providers) configuration.
